### PR TITLE
Add queries to database

### PR DESCRIPTION
**Pull request by niicode**
In this milestone I was able to achieve the following:
- Query and Update the vet_clinic database, below are screenshots of my queries.
[screenshot 1](https://github.com/niicode/vet_clinic/blob/queries/screenshots/Screenshot%202022-10-19%20at%2010.12.25%20AM.pngl)
[screenshot 2](https://github.com/niicode/vet_clinic/blob/queries/screenshots/Screenshot%202022-10-19%20at%2010.13.15%20AM.png)
[screenshot 3](https://github.com/niicode/vet_clinic/blob/queries/screenshots/Screenshot%202022-10-19%20at%2010.14.09%20AM.png)
[screenshot 4](https://github.com/niicode/vet_clinic/blob/queries/screenshots/Screenshot%202022-10-19%20at%2010.14.55%20AM.png)

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -4,3 +4,11 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
 INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES (2, 'Gabumon', '2018-11-15', 2, TRUE, 8.0);
 INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES (3, 'Pikachu', '2021-01-07', 1, FALSE, 15.0);
 INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES (4, 'Devimon', '2017-05-12', 5, TRUE, 11.0);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Charmander', date '2020-02-8', '0', 'true', '-11');
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Plantmon', date '2021-10-15', '2', 'true', '-5.7');
```
Comment (rustam-tolipov) [inline data.sql]: Kindly insert correct date `2021-10-15` incorrect.  `2021-11-15` correct.

File: data.sql [hunk 2]
```diff
@@ -4,3 +4,11 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
...
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Squirtle', date '1993-04-03', '3', 'false', '-12.13');
```
Comment (rustam-tolipov) [inline data.sql]: Not `1993-04-03`. Correct `1993-04-02`

File: data.sql [hunk 3]
```diff
@@ -4,3 +4,11 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
...
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Angemon', date '2005-06-12', '1', 'true', '-45');
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Boarmon', date '2005-06-07', '7', 'true', '20.4');
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Blossom', date '1998-08-13', '3', 'true', '17');
```
Comment (rustam-tolipov) [inline data.sql]: Not `1998-08-13`. Correct `1998-10-13`

File: queries.sql [hunk 4]
```diff
@@ -23,3 +23,32 @@ SELECT * FROM animals WHERE name != 'Gabumon';
 
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg).
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
```
Comment (rustam-tolipov) [inline queries.sql:29]: - [ ] You are missing two queries. 
1. You must select species from animals and rollback
2. After rollback you need to make sure you selected species from animals again.

File: queries.sql [hunk 5]
```diff
@@ -23,3 +23,32 @@ SELECT * FROM animals WHERE name != 'Gabumon';
...
+ROLLBACK;
+
+BEGIN;
+UPDATE animals SET species = 'Digmon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'Pokemon' WHERE species IS NULL;
+COMMIT;
+
+BEGIN;
+DELETE FROM animals;
```
Comment (rustam-tolipov) [inline queries.sql:38]: After deleting. `COUNT(*)` them from animals then you can rollback again.

File: queries.sql [hunk 6]
```diff
@@ -23,3 +23,32 @@ SELECT * FROM animals WHERE name != 'Gabumon';
...
+ROLLBACK;
+
```
Comment (rustam-tolipov) [inline queries.sql:40]: - [ ] After rollback make sure you `COUNT(*)` them again.

File: queries.sql [hunk 7]
```diff
@@ -23,3 +23,32 @@ SELECT * FROM animals WHERE name != 'Gabumon';
...
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO sp1;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT MAX(escape_attempts) FROM animals;
+SELECT MIN(weight_kg), MAX(weight_kg) FROM animals;
```
Comment (rustam-tolipov) [inline queries.sql:53]: - [ ] Kindly make `MAX` and `MIN` in one query.

File: queries.sql [hunk 8]
```diff
@@ -23,3 +23,32 @@ SELECT * FROM animals WHERE name != 'Gabumon';
...
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT sp1;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO sp1;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
```
Comment (rustam-tolipov) [inline queries.sql:51]: - [ ] You also need to write query to check `AVG` of neutered animals separately as well.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @niicode, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_
_Great Design_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 4 above)
Comment (niicode) [reply queries.sql:29]: @Rustamxon7 This was not part of the requirements please, check to verify

Comment (rustam-tolipov) [reply queries.sql:29]: Let's look through them one by one.

Screen Shot 2022-10-19 at 16 55 57

File: queries.sql (hunk 5 above)
Comment (rustam-tolipov) [reply queries.sql:38]: Screen Shot 2022-10-19 at 17 01 15

Comment (niicode) [reply queries.sql:38]: @Rustamxon7 I read these instructions and verified them, I am sure you know that if they didn't work some of the queries I made would have been wrong, I am not sure if I was supposed to take screenshots of all verifications.

File: queries.sql (hunk 8 above)
Comment (rustam-tolipov) [reply queries.sql:51]: You can make it AVG or Max or Min

Screen Shot 2022-10-19 at 17 04 08

Comment (niicode) [reply queries.sql:51]: @Rustamxon7 Thanks for your feedback I will work on all your detailed changed

File: queries.sql (hunk 7 above)
Comment (Rustamxon7) [reply queries.sql:53]: I would make it two queries if there were `,` before `and`.

Screen Shot 2022-10-19 at 17 06 35

File: queries.sql (hunk 5 above)
Comment (rustam-tolipov) [reply queries.sql:38]: Yes Exactly, you did a great job. However, these are the requirements that shown to us.

File: queries.sql (hunk 7 above)
Comment (rustam-tolipov) [reply queries.sql:53]: It's one requirement and not two. Also, I would make it two queries if there were `,` before `and`.

Screen Shot 2022-10-19 at 17 06 35

**Review by alexander16108 — APPROVED** [review]
Hi @niicode , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**